samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -230,6 +243,162 @@ static void location_evt_led_worker(struct k_work *work_item)
 
 /******************************************************************************/
 
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
+static void location_ctrl_agps_ext_handle(const struct nrf_modem_gnss_agps_data_frame *agps_req)
+{
+	int err;
+
+	if (!cloud_lwm2m_is_connected()) {
+		mosh_error("LwM2M not connected, can't request A-GPS data");
+		return;
+	}
+
+	location_assistance_agps_set_mask(agps_req);
+
+	while ((err = location_assistance_agps_request_send(cloud_lwm2m_client_ctx_get(), true)) ==
+	       -EAGAIN) {
+		/* LwM2M client utils library is currently handling a P-GPS data request, need to
+		 * wait until it has been completed.
+		 */
+		k_sleep(K_SECONDS(1));
+	}
+	if (err) {
+		mosh_error("Failed to request A-GPS data, err: %d", err);
+	}
+}
+#endif
+
+#if defined(CONFIG_LOCATION_METHOD_GNSS_PGPS_EXTERNAL)
+static void location_ctrl_pgps_ext_handle(const struct gps_pgps_request *pgps_req)
+{
+	int err = -1;
+
+	if (!cloud_lwm2m_is_connected()) {
+		mosh_error("LwM2M not connected, can't request P-GPS data");
+		return;
+	}
+
+	err = location_assist_pgps_set_prediction_count(pgps_req->prediction_count);
+	err |= location_assist_pgps_set_prediction_interval(pgps_req->prediction_period_min);
+	location_assist_pgps_set_start_gps_day(pgps_req->gps_day);
+	err |= location_assist_pgps_set_start_time(pgps_req->gps_time_of_day);
+	if (err) {
+		mosh_error("Failed to set P-GPS request parameters");
+		return;
+	}
+
+	while ((err = location_assistance_pgps_request_send(cloud_lwm2m_client_ctx_get(), true)) ==
+	       -EAGAIN) {
+		/* LwM2M client utils library is currently handling an A-GPS data request, need to
+		 * wait until it has been completed.
+		 */
+		k_sleep(K_SECONDS(1));
+	}
+	if (err) {
+		mosh_error("Failed to request P-GPS data, err: %d", err);
+	}
+}
+#endif
+
+#if defined(CONFIG_LOCATION_METHOD_CELLULAR_EXTERNAL)
+static int location_ctrl_ground_fix_result_cb(uint16_t obj_inst_id,
```

Comment:
Guys have been told me many times that we shall not use word "fix" in other than GNSS context  ;-)
Yeah, in this case, it comes from lwm2m. But please add at least lwm2m word here as this is just with lwm2m.